### ISSUE TITLE
Chain of trust problem

### ISSUE THREAD

**Issue by drwetter** [drwetter/testssl.sh#278]
Hi @dcooper16.

it seems that ``openssl verify`` is not as reliable as just doing a ``openssl s_client connect`` and then look out for the return code  / verify (ok).

I have a problem with a self signed certificate which is not marked as such in the "Chain of trust' (try localhost:631 if you're running CUPS) And e.g. for the host ``testssl.sh`` the Java keystore fails, thus marking the whole site as not trusted.

Unless you have a better idea I would revert that part back to the old check.

Cheers, Dirk

**Comment by drwetter** [comment]
Partly SSL_CERT_DIR seems to be the "culprit", at least for testssl.sh and the Java store. 

Where do you have that variable from? In strace I see that the system store is being used the root StartCom certificate is being added if SSL_CERT_DIR is not set to /dev/null. That root CA doesn't seem to be in java.pem.

So there I guess should probably remove the java store.

For the second case it looks the ret code is not properly parsed:

```
mylaptop:/tmp/ssltester.J3VrVa|% ~/testssl/bin/openssl.Linux.x86_64 verify -purpose sslserver -CAfile ~/testssl/etc/java.pem  host_certificate.txt 
host_certificate.txt: CN = <myselfsignedhost-CN>
error 18 at 0 depth lookup:self signed certificate
OK
mylaptop:/tmp/ssltester.J3VrVa|% echo $?                                                     
mylaptop:/tmp/ssltester.J3VrVa|%           
```

**Comment by dcooper16** [comment]
Okay, it seems that for the verify command, I need to parse the text that is output by the command rather than relying on the return code.

Most of the time when I've been running my tests I've only had a single file with my own locally-generated root certificate in $INSTALL_DIR/etc/*.pem, but I noticed that "Chain of trust" always said "Ok" for any public servers that I tried. So, I knew that OpenSSL had to be using a system store of trusted certificates in addition to those provided by the "-CAfile" option. After searching through the source code I found it was using SSL_CERT_DIR as the local of the system store and setting that to /dev/null forced OpenSSL to only use the trusted certificates included in "-CAfile"
